tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -1013,6 +1181,11 @@ func (az *AzureProvider) DestroyWindowsVM(vmName string) error {
 		log.Printf("error destroying the instance %s: %v", vmName, err)
 	}
 
+	err = az.disassociateIP(ctx, nicName)
+	if err != nil {
+		log.Printf("error disassociating the Public IP of instance %s: %v", vmName, err)
+	}
+
```

Comment:
Do we need to wait until we are sure the disassociation is complete as it not being complete can cause `destroyIP()` to fail?

Reply:
API handles that and will return an error if it fails to disassociate it of if it is unable to disassociate it

Reply:
Sure but would it help if we waited? Otherwise we are going to end up leaking resources.

Reply:
added polling in disassociate function.